simulation/src/io/WriterHDF5.cpp

```
@@ -0,0 +1,438 @@
+            xdmfFile << "<DataItem";
+            xdmfFile << " NumberType=" << '"' << "Float" << '"';
+            xdmfFile << " Precision=" << '"' << precision << '"';
+
+            xdmfFile << " Dimensions=\"";
+            for (int idim = 2; idim >= 0; --idim) {
+                xdmfFile << ncells[idim];
+                xdmfFile << (idim == 0 ? "\"" : " ");
+            }
+
+            xdmfFile << " Format=" << '"' << "HDF" << '"';
+            xdmfFile << ">\n";
+            xdmfFile << std::string(12, ' ') << getFilename(it.first)
+                        << ":/" << var_name << "\n";
+            xdmfFile << std::string(10, ' ') << "</DataItem>\n";
+            xdmfFile << std::string(8, ' ') << "</Attribute>\n";
+        }
+        // finalize grid file for the current time step
+        xdmfFile << std::string(6, ' ') << "</Grid>\n";
+    }
+
+    // finalize Xdmf wrapper file
+    xdmfFile << std::string(4, ' ') << "</Grid>\n";
+    xdmfFile << std::string(2, ' ') << "</Domain>\n";
+    xdmfFile << std::string(0, ' ') << "</Xdmf>\n";
+}
+
+
+void WriterHDF5::write(HostConstArrayDyn u, const UniformGrid& grid, 
+                            Int iStep, Real time, Real gamma, Real mmw)
+{
+    auto& outputId = WriterBase::m_outputId;
+    const auto& restartId = WriterBase::m_restartId;
+    
+    std::string filename = getFilename(outputId);
+    const auto& Rstar_h = code_units::constants::Rstar_h;
+
+    // Jacques:::: A voir si on met la version local
+    std::array<int, 3> ncells;
+    ncells[IX] = grid.m_nbCells[IX] * grid.m_dom[IX];
+    ncells[IY] = grid.m_nbCells[IY] * grid.m_dom[IY];
+    ncells[IZ] = grid.m_nbCells[IZ] * grid.m_dom[IZ];
+
+    const int dim = three_d;
+
+    // create file 
+    hid_t file_id = H5Fcreate(filename.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
+	if (file_id < 0) {
+        std::cout << "Error "<< file_id << " in H5Fcreate for filename=" << filename << std::endl;
+		exit(1);
```

could we throw an error here?